ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -381,10 +387,36 @@ internal constructor(
   fun isLastPosition(taskId: String? = null): Boolean =
     (taskId ?: currentTaskId.value) == getTaskSequence().last().id
 
+  /**
+   * Returns true if the given [taskId] and task data would be last, or the current active task.
+   * Useful for handling conditional tasks.
+   */
+  fun testLastPosition(taskId: String? = null, value: TaskData?): Boolean =
```

Comment:
is it testing the last position, or taskLastPosition or isTaskOnLastPosition?

Reply:
Renamed to be slightly more clear, `checkLastPositionWithTaskData()`!